app/services/PaymentHistoryService.scala

```diff
@@ -42,7 +43,8 @@ class PaymentHistoryService @Inject()(repaymentHistoryConnector: RepaymentHistor
     Future.sequence(orderedTaxYears.map(financialDetailsConnector.getPayments)) map { paymentResponses =>
       val paymentsContainsFailure: Boolean = paymentResponses.exists {
         case Payments(_) => false
-        case PaymentsError(status, _) if status == 404 => false
+        case PaymentsError(status, _) if status == NOT_FOUND => false
+        case PaymentsError(status, _) if status == UNPROCESSABLE_ENTITY => true
```

Comment:
there is no logical changes applied after line 47 was added
line 48 will always return true, so line 47 is not required